src/Controller/CRUDController.php

```diff
@@ -646,16 +649,18 @@ public function createAction()
 
             // show an error message if the form failed validation
             if (!$isFormValid) {
-                if (!$this->isXmlHttpRequest()) {
-                    $this->addFlash(
-                        'sonata_flash_error',
-                        $this->trans(
-                            'flash_create_error',
-                            ['%name%' => $this->escapeHtml($this->admin->toString($newObject))],
-                            'SonataAdminBundle'
-                        )
-                    );
+                if ($this->isXmlHttpRequest()) {
```

Comment:
I know there are no checks for this currently, but if we decide to return JSON responses when the request is made from a XHR, we could check the "accept" header before.

Reply:
That could be a good improvement but I think it could be a B/C break because not all clients will be configured correctly. How about creating a separate PR?

Reply:
Maybe I'm missing something, but AFAIK, this method is not not capable to return a JSON response without these changes, so I think it will not be a BC break.

Reply:
It actually was returning JSON but only for success
https://github.com/sonata-project/SonataAdminBundle/blob/2a5da3588e972abcb4a1721e891598ac03668801/src/Controller/CRUDController.php#L359